things/redis/groups.go

```diff
@@ -0,0 +1,181 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/MainfluxLabs/mainflux/pkg/errors"
+	"github.com/MainfluxLabs/mainflux/things"
+	"github.com/go-redis/redis/v8"
+)
+
+const (
+	groupOrgPrefix     = "gr_org"
+	groupMembersPrefix = "gr_mbs"
+	memberGroupsPrefix = "mb_grs"
+)
+
+var _ things.GroupCache = (*groupCache)(nil)
+
+type groupCache struct {
+	client  *redis.Client
+	thCache things.ThingCache
+}
+
+// NewGroupCache returns redis group cache implementation.
+func NewGroupCache(client *redis.Client, thCache things.ThingCache) things.GroupCache {
+	return &groupCache{
+		client:  client,
+		thCache: thCache,
+	}
+}
+
+func (gc *groupCache) SaveOrg(ctx context.Context, groupID, orgID string) error {
+	gok := groupOrgKey(groupID)
+	if err := gc.client.Set(ctx, gok, orgID, 0).Err(); err != nil {
+		return errors.Wrap(errors.ErrCreateEntity, err)
+	}
+
+	return nil
+}
+
+func (gc *groupCache) ViewOrg(ctx context.Context, groupID string) (string, error) {
+	gok := groupOrgKey(groupID)
+	orgID, err := gc.client.Get(ctx, gok).Result()
+	if err != nil {
+		return "", errors.Wrap(errors.ErrNotFound, err)
+	}
+
+	return orgID, nil
+}
+
+func (gc *groupCache) RemoveOrg(ctx context.Context, groupID string) error {
+	var (
+		cursor uint64
+		keys   []string
+		err    error
+	)
+
+	if err = gc.clearGroup(ctx, groupID, groupThingsPrefix, groupProfilesPrefix, groupMembersPrefix); err != nil {
+		return err
+	}
+
+	// Using this key pattern will delete keys whose prefix starts with 'gr' and the suffix is '<group_id>', such as:
+	// gr_org:<group_id>, gr_ths:<group_id>, gr_prs:<group_id>, gr_mbs:<group_id>
+	key := fmt.Sprintf("gr_*:%s", groupID)
```

Comment:
It would be more readable and we would avoid `Scan` by using `keys=[groupOrgPrefix, groupMembersPrefix, membersGroupPrefix ]` and range over it